source/client/service_main.cc

```diff
@@ -20,37 +20,44 @@ ServiceMain::ServiceMain(int argc, const char** argv) {
       "The address:port on which the Nighthawk grpc service should listen. Default: "
       "0.0.0.0:8443.",
       false, "0.0.0.0:8443", "address:port", cmd);
-
   Utility::parseCommand(cmd, argc, argv);
 
-  try {
-    listener_address_ =
-        Envoy::Network::Utility::parseInternetAddressAndPort(internet_address_and_port.getValue());
-  } catch (Envoy::EnvoyException e) {
-    throw NighthawkException(e.what());
+  parseIpAndMaybePort(internet_address_and_port.getValue());
+  ENVOY_LOG(info, "Nighthawk grpc service listener binding to: {}", getListenerAddress());
+  builder_.AddListeningPort(getListenerAddress(), grpc::InsecureServerCredentials(), &port_);
+  builder_.RegisterService(&service_);
+}
+
+// TODO(oschaaf): move to utility class.
+void ServiceMain::parseIpAndMaybePort(absl::string_view ip_and_maybe_port) {
```

Comment:
I would much prefer to have these utilities in Envoy base (and ideally built on existing things in `Envoy::Network::Utility` and `Address::Instance`. It's pretty fidgetty doing all of this, and ideally we do it once in the Envoy world.

Reply:
I updated this to restrict the parsing to a minimum